File: leetcode_cpp/unique_binary_search_trees.cpp
class Solution {
public:
    int numTrees(int n) {
        if (n <= 1) {
            return 1;
        }
        int sum = 0, sum_left, sum_right;
        for (int i = 1; i <= n; ++i) {
            sum_left = numTrees(i - 1);
            sum_right = numTrees(n - i);
            sum += sum_left * sum_right;
        }
        return sum;
    }
};

File: leetcode_python/unique_binary_search_trees.py
class Solution:
    # @return an integer
    def numTrees(self, n):
        return 1 if n <= 1 else sum(self.numTrees(i) * self.numTrees(n - 1 - i) for i in range(0, n))

# iterate through all possible root nodes, split into left and right subtrees.
